### PR TITLE
Added example to CP.61: Use an async() to spawn a concurrent task

### PR DESCRIPTION
**Pull request by roroberto**
Added example to CP.61: Use an async() to spawn a concurrent task

### CODE REVIEW

File: CppCoreGuidelines.md [hunk 1]
```diff
@@ -14924,8 +14924,38 @@ There is no explicit locking and both correct (value) return and error (exceptio
...
+    void async_example()
+    {
+        auto f = std::async(std::launch::async, fun, 10);
 
+        std::cout << f.get() << '\n';
+
+        try
+        {
+            auto fe = std::async(std::launch::async, fun_ex);
+            auto i = fe.get(); // this call will thow a Moonphase_error exception
```
Comment (cubbimew) [inline CppCoreGuidelines.md]: s/thow/throw
but also, this example seems to take a lot of real estate to demo custom error handling, but shows no concurrent execution. How about something more like `parallel_sum` from  https://en.cppreference.com/w/cpp/thread/async#Example

**Comment by hsutter** [comment]
Editors call: Assigned to Sergey, thanks!

File: CppCoreGuidelines.md (hunk 1 above)
Comment (roroberto) [reply CppCoreGuidelines.md]: ok, thank you. I see your point.
Give me a little bit to think about it.

Comment (roroberto) [reply CppCoreGuidelines.md]: I have committed another example.
Please let me know what you think about it.

**Pull request merged**